[FRNT-520] Rewrite tooltip (#129)
woly-ui/woly: src/ui/atoms/tooltip/usage.mdx

---

import {block, Playground} from 'lib/playground'
import {Button, Input, Tooltip, Text} from 'ui'
import {IconInfo} from 'static/icons';

> `Tooltip` is used to show addition information near component. The information should be contextual, useful, and nonessential.
> 
> A tooltip can be simple message box that is displayed when a user hovers over, clicks or gives focus to a UI element
> such as an icon, a highlighted word, or a button that is passed as a children to `Tooltip`.
> Also the definition tooltip provides additional help or defines an item or term. It may be used on the label of a UI element,
> or on a word embedded in a paragraph.
> 
> Interactive tooltips may contain rich text and other interactive elements like buttons or links.
> In general, hiding interactive content in a tooltip is discouraged.
> Interactive tooltips are best used for onboarding experiences and product tours.
> 
> The tooltip position is flexible and changes depending on how close the element is to the edge of the screen.
> The top, right, bottom, and left positions appear outside the target element, center-aligned along the appropriate axis.

Reviewer: ```suggestion
`Tooltip` is used to show additional information near the component. The information should be contextual, useful, and nonessential.

A tooltip can be a simple message box that is displayed when a user hovers over, clicks, or gives focus to a UI element
such as an icon, a highlighted word, or a button that is passed as children to `Tooltip`.
Also, the definition tooltip provides additional help or defines an item or term. It may be used on the label of a UI element,
or on a word embedded in a paragraph.

Interactive tooltips may contain rich text and other interactive elements like buttons or links.
In general, hiding interactive content in a tooltip is discouraged.
Interactive tooltips are best used for onboarding experiences and product tours.

The tooltip position is flexible and changes depending on how close the element is to the edge of the screen.
The top, right, bottom, and left positions appear outside the target element, center-aligned along the appropriate axis.
```

### Example

<Playground>
  <block.S
    style={{
      width: '800px',
      display: 'flex',
      padding: '200px',
      alignItems: 'center',
      justifyContent: 'space-between',
      background: 'grey',
    }}
  >

>     <Tooltip
>       text={<Text type="S">Подсказка может быть написана в две строки и более</Text>}
>       position="bottom"
>     >
>       <Button onClick={() => console.info('Hi!')} text="Button" variant="primary" />
>     </Tooltip>

Reviewer 2: Давай переименуем text в children, а действия в actions?
А то сейчас выглядит, будто бы кнопки это основной контент, а текст вспомогательный.

```suggestion
    <Tooltip
      actions={<Button onClick={() => console.info('Hi!')} text="Button" variant="primary" />}
      position="bottom"
    >
      <Text type="S">Подсказка может быть написана в две строки и более</Text>
    </Tooltip>
```

Reviewer 2: Думаю, что чаще будут тултипы с текстом без кнопок

### Example

<Playground>
  <block.S
    style={{
      width: '800px',
      display: 'flex',
      padding: '200px',
      alignItems: 'center',
      justifyContent: 'space-between',
      background: 'grey',
    }}
  >
    <Tooltip
      text="Подсказка может быть написана в две строки и более"
      position="bottom"
      isOpen={false}
    >
      <Button onClick={() => console.info('Hi!')} text="Button" variant="primary" />
    </Tooltip>
    <Tooltip text="Подсказка" position="top" id="tooltip-top" isOpen={false}>
      <Button onClick={() => console.info('Hi!')} text="Button" variant="primary" />
    </Tooltip>
    <Tooltip
      text="Подсказка может быть написана в две и более строки, а также содержать блок с actions"
      position="left"
      isOpen={false}
      iconRight={
        <>
          <Button onClick={() => console.info('Hi!')} text="Button" variant="primary" />
          <Button onClick={() => console.info('Hi!')} text="Button" variant="secondary" />
        </>
      }
    >
      <Button onClick={() => console.info('Hi!')} text="Button" variant="primary" />
    </Tooltip>

>     <Tooltip text="Подсказка" position="right" variant="primary" isOpen={false}>

Reviewer: Пропсы поменялись, а в внизу в разделе Props не изменились

### Props

> | Name        | Type                 | Default       | Description                                                                              |
> | ----------- | -------------------- | ------------- | ---------------------------------------------------------------------------------------- |
> | `children`  | `React.ReactElement` |               | The tooltip will be visible near `children` on hover or focus                            |
> | `column`    | `boolean`            | `false`       | Column prop set deriction of the tooltip content                                         |
> | `iconLeft`  | `React.ReactNode`    | `null`        | Component to show on the left side of the description (ex.: icon)                        |
> | `iconRight` | `React.ReactNode`    | `null`        | Component to show on the right side of the description (ex.: icon or block with actions) |
> | `position`  | `string`             | `'top'`       | Positon of tooltip aginst the relative element                                           |
> | `text`      | `React.ReactElement` |               | Information has been described in tooltip                                                |
> | `variant`   | `string`             | `'secondary'` | Variant prop to style Tooltip component                                                  |

Reviewer: ```suggestion
| Name        | Type                 | Default       | Description                                                                              |
| ----------- | -------------------- | ------------- | ---------------------------------------------------------------------------------------- |
| `children`  | `React.ReactElement` |               | The tooltip will be visible near `children` on hover or focus                            |
| `column`    | `boolean`            | `false`       | Column prop set the direction of the tooltip content                                     |
| `iconLeft`  | `React.ReactNode`    | `null`        | Component to show on the left side of the description (ex.: icon)                        |
| `iconRight` | `React.ReactNode`    | `null`        | Component to show on the right side of the description (ex.: icon or block with actions) |
| `position`  | `string`             | `'top'`       | The position of the tooltip against the relative element                                 |
| `text`      | `React.ReactElement` |               | Information has been described in a tooltip                                              |
| `variant`   | `string`             | `'secondary'` | Variant prop to style Tooltip component                                                  |                                           |
```
